### DISCUSSION TITLE
Netlify deployment issues

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#266] [Help Me!] [answered]
I have had issues trying to deploy my react bio site for days with no success. I have tried everything listed in the following discussion tickets but no luck:
#248 
#262
#263 
#230 

I have followed all the steps in setting up the react template and instituted all the rules for deployment that were listed in the React Template README.md but am still getting the errors listed below.

Attached is the relevant portion of my Netlify deploy log:
9:57:17 PM: ​
9:57:17 PM: $ npm run build
9:57:17 PM: > react-template@0.1.0 build /opt/build/repo
9:57:17 PM: > react-scripts build
9:57:19 PM: Creating an optimized production build...
9:57:24 PM: Failed to compile.
9:57:24 PM: 
9:57:24 PM: ./src/App/index.js
9:57:24 PM: Cannot find file '../helpers/data/ProjectsData' in './src/App'.
9:57:24 PM: npm ERR! code ELIFECYCLE
9:57:24 PM: npm ERR! errno 1
9:57:24 PM: npm ERR! react-template@0.1.0 build: `react-scripts build`
9:57:24 PM: npm ERR! Exit status 1
9:57:24 PM: npm ERR!
9:57:24 PM: npm ERR! Failed at the react-template@0.1.0 build script.
9:57:24 PM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
9:57:24 PM: npm ERR! A complete log of this run can be found in:
9:57:24 PM: npm ERR!     /opt/buildhome/.npm/_logs/2021-06-01T02_57_24_427Z-debug.log
9:57:24 PM: ​
9:57:24 PM: ────────────────────────────────────────────────────────────────
9:57:24 PM:   "build.command" failed                                        
9:57:24 PM: ────────────────────────────────────────────────────────────────
9:57:24 PM: ​
9:57:24 PM:   Error message
9:57:24 PM:   Command failed with exit code 1: npm run build
9:57:24 PM: ​
9:57:24 PM:   Error location
9:57:24 PM:   In Build command from Netlify app:
9:57:24 PM:   npm run build
9:57:24 PM: ​
9:57:24 PM:   Resolved config
9:57:24 PM:   build:
9:57:24 PM:     command: npm run build
9:57:24 PM:     commandOrigin: ui
9:57:24 PM:     environment:
9:57:24 PM:       - REACT_APP_ADMIN_UID
9:57:24 PM:       - REACT_APP_API_KEY
9:57:24 PM:       - REACT_APP_APP_ID
9:57:24 PM:       - REACT_APP_AUTH_DOMAIN
9:57:24 PM:       - REACT_APP_DATABASE_URL
9:57:24 PM:       - REACT_APP_MEASUREMENT_ID
9:57:24 PM:       - REACT_APP_MESSAGING_SENDER_ID
9:57:24 PM:       - REACT_APP_PROJECT_ID
9:57:24 PM:       - REACT_APP_STORAGE_BUCKET
9:57:24 PM:     publish: /opt/build/repo/build
9:57:24 PM:   functions:
9:57:24 PM:     '*': {}
9:57:24 PM: Caching artifacts
9:57:24 PM: Started saving node modules
9:57:24 PM: Finished saving node modules
9:57:24 PM: Started saving build plugins
9:57:24 PM: Finished saving build plugins
9:57:24 PM: Started saving yarn cache
9:57:24 PM: Finished saving yarn cache
9:57:24 PM: Started saving pip cache
9:57:24 PM: Finished saving pip cache
9:57:24 PM: Started saving emacs cask dependencies
9:57:24 PM: Finished saving emacs cask dependencies
9:57:24 PM: Started saving maven dependencies
9:57:24 PM: Finished saving maven dependencies
9:57:24 PM: Started saving boot dependencies
9:57:25 PM: Finished saving boot dependencies
9:57:25 PM: Started saving rust rustup cache
9:57:25 PM: Finished saving rust rustup cache
9:57:25 PM: Started saving go dependencies
9:57:25 PM: Finished saving go dependencies
9:57:27 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
9:57:27 PM: Creating deploy upload records
9:57:27 PM: Failing build: Failed to build site
9:57:27 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2
9:57:27 PM: Finished processing build request in 1m15.256231905s

My file structure:
![Screen Shot 2021-05-31 at 10 38 10 PM](https://user-images.githubusercontent.com/76710923/120263339-1ed7af80-c261-11eb-9f8f-6863c29c8857.png)

Link to GitHub repo: https://github.com/nashvegasnate/react-bio-site

**Comment by nashvegasnate** [comment, marked as answer]
UPDATE: I figured it out. My ProjectsData.js file was properly labeled in my VSCode locally, however, it was listed as projectsData.js in my repo on GitHub. Not sure why it was ever listed that way and despite several saves and pushes to the repo, it never changed.  So, I had to go onto GitHub and do a manual edit to change it so that it reflected the necessary capitalization as ProjectsData.js.

**Reply by thedigitalmenagerie** [reply]
How did you manually rename files in GitHub? I think I am having the same issue and need to get deploy going so I can have more users for testing during the duration of Capstones!

**Comment by nashvegasnate** [comment]
*Hey! How you change the name of the file is to pull up the code you want
to change in GItHub and then click on the pencil in the upper right hand
corner. *

[image: Screen Shot 2021-06-14 at 6.53.35 PM.png]

*Then, when the code pops up and is ready to edit, go up to the branch at
the top left hand corner above the code and in the text input field where
it shows the file name, just manually edit the name of the file and save
the changes. *



*I hope that helps!*

[image: Screen Shot 2021-06-14 at 6.49.15 PM.png]
